packages/app/src/applications/multi-instance-configuration/webui/components/ChooseIdentityForm.tsx

```diff
@@ -49,18 +46,20 @@ export default class ChooseIdentityForm extends React.PureComponent<Props, State
 
   // tslint:disable-next-line:function-name
   UNSAFE_componentWillMount() {
-    this.subscription = window.bx.identities.$get.subscribe(identities => {
+    window.bx.identities.addIdentitiesChangeListener((_: any, identities: any) => {
       this.setState({
         identities,
       });
     });
   }
 
-  componentWillUnmount() {
-    if (this.subscription) {
-      this.subscription.unsubscribe();
-    }
-  }
+  //vk: FIXME: should we remove listener? how?
+
+  // componentWillUnmount() {
+  //   if (this.subscription) {
+  //     this.subscription.unsubscribe();
+  //   }
+  // }
```

Comment:
I think that we should by default without thinking too much about it, it avoids potential memory leaks.
Perhaps all functions on `window.bx` should return an unsubscriber, usable like so?
```ts
const unsub = window.bx.identities.addIdentitiesChangeListener((_: any, identities: any) => {
  this.setState({
    identities,
  });
});

componentWillUnmount() {
  unsub();
}
```

Reply:
fixed